src/library/coverartcache.cpp

```diff
@@ -158,10 +158,12 @@ void CoverArtCache::tryLoadCover(
     }
 
     const auto requestedCacheKey = coverInfo.cacheKey();
-    // keep a list of trackIds for which a future is currently running
+    QString cacheKey = pixmapCacheKey(requestedCacheKey, desiredWidth);
+    // keep a list of cacheKeys for which a future is currently running
     // to avoid loading the same picture again while we are loading it
-    QPair<const QObject*, mixxx::cache_key_t> requestId = qMakePair(pRequester, requestedCacheKey);
-    if (m_runningRequests.contains(requestId)) {
+    bool requestPending = m_runningRequests.contains(cacheKey);
```

Comment:
Here we collect requests to a same file. 
Whe the first look up is done m_runningRequests is used to signall all requesters.

Reply:
Got it! Thanks, this helps!